flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlDdlToOperationConverterTest.java

```diff
@@ -804,6 +804,61 @@ public void testCreateTableAsWithColumnsOverridden() {
                                                         .build()))));
     }
 
+    @Test
+    public void testCreateTableAsWithOverriddenMetadataColumns() {
+        CatalogTable catalogTable =
+                CatalogTable.newBuilder()
+                        .schema(
+                                Schema.newBuilder()
+                                        .column("f0", DataTypes.INT().notNull())
+                                        .column("f1", DataTypes.BIGINT())
+                                        .build())
+                        .build();
+
+        catalogManager.createTable(
+                catalogTable, ObjectIdentifier.of("builtin", "default", "src1"), false);
+
+        final String sql = "create table tbl1 (f1 bigint metadata) " + "AS SELECT * FROM src1";
```

Comment:
I am wondering if we could test: 
table src2 g1 and g2

`create table tbl1 (f1 bigint metadata)  AS SELECT s.g1 as f1 FROM src2 as s;`

and is this sort of thing valid - or as the 2 ASes not allowed?
`create table tbl1 (g1 bigint metadata as f1)  AS SELECT s.g1 FROM src2 as s;`

Reply:
The second case  is not valid. You cannot have aliases in the CREATE part, only in the SELECT part. I updated a test to include aliases on physical and metadata columns. The same for RTAS tests.